src/MongoDB.Driver.Core/Core/Clusters/CryptClientCreator.cs

```diff
@@ -105,7 +97,18 @@ private CryptOptions CreateCryptOptions()
                 schemaBytes = schemaDocument.ToBson(writerSettings: writerSettings);
             }
 
-            return new CryptOptions(kmsProvidersMap, schemaBytes);
+            return new CryptOptions(kmsProviders, schemaBytes);
+        }
+
+        private BsonDocument CreateProviderDocument(string kmsType, IReadOnlyDictionary<string, object> data)
+        {
+            var providerContent = new BsonDocument();
+            foreach (var record in data)
+            {
+                providerContent.Add(new BsonElement(record.Key, BsonValue.Create(record.Value)));
```

Comment:
Calling `BsonValue.Create` with an `object` argument is dangerous.

Do you know what values for `record.Value` might appear here?

Reply:
any which was provided by user, but we expect only `byte[]` and `string`

Reply:
discussed offline